temporian/core/data/node.py

```diff
@@ -369,84 +562,87 @@ def __and__(self, other: Any) -> Node:
             from temporian.core.operators.binary import logical_and
 
             return logical_and(input_1=self, input_2=other)
-        self._nope_only_boolean("&", other)
+
+        self._error_only_boolean("&", other)
+        assert False
 
     def __or__(self, other: Any) -> Node:
         if isinstance(other, Node):
             from temporian.core.operators.binary import logical_or
 
             return logical_or(input_1=self, input_2=other)
-        self._nope_only_boolean("|", other)
+
+        self._error_only_boolean("|", other)
+        assert False
 
     def __xor__(self, other: Any) -> Node:
         if isinstance(other, Node):
             from temporian.core.operators.binary import logical_xor
 
             return logical_xor(input_1=self, input_2=other)
-        self._nope_only_boolean("^", other)
 
-    @property
-    def sampling(self) -> Sampling:
-        return self._sampling
-
-    @property
-    def features(self) -> List[Feature]:
-        return self._features
+        self._error_only_boolean("^", other)
+        assert False
 
-    @property
-    def feature_names(self) -> List[str]:
-        return [feature.name for feature in self._features]
 
-    @property
-    def index_names(self) -> List[str]:
-        return self.sampling.index.names
+def input_node(
```

Comment:
could we standardize this name? either by defining it as `source_node` here or by exposing `input_node` in the public API. I personally think `input` is clearer but both work

Reply:
The "source" concept is related to graphs, so I like "source node". Let me unify it.